### ISSUE TITLE
Modal dialog is not correctly annunced for screen readers and is not returning the focus to the button/link trigger

### ISSUE THREAD

**Issue by marcoscv-work** [liferay/clay#5295]
**Context**:
Clay modal: When we activate a button to open a modal we are moving the focus to the modal (`role="dialog"`) and SR reads immediately the `aria-labelledby=""` that is the modal title.
When we close the modal the focus is lost or at the top of the page

**Issue**: 
- It should annunce that dialog is already opened so you are in a dialog
- When we close the modal the focus have to go to the trigger witch opened the modal
- Modal title should be H1
- We could need to change (`role="dialog"`) by (`role="alertdialog"`)

**How to fix / recommendation**:
- I've seen bootstrap is working in a really similar way by using `aria-labelledby=""` but everithing is being annunced correctly:

![image](https://user-images.githubusercontent.com/7963804/212088209-4e6a9674-e2f6-426b-a38a-3aa8dadd063f.png)
https://getbootstrap.com/docs/5.3/components/modal/#live-demo

Could we apply the bootstrap pattern?

- About the H1, just addit to `modal-title`
- About the focus on close, please move te focus to the rigger witch opened the modal

---
Issues comming from DXP:
https://issues.liferay.com/browse/LPS-168629
https://issues.liferay.com/browse/LPS-168533
https://issues.liferay.com/browse/LPS-168402
https://issues.liferay.com/browse/LPS-168512
https://issues.liferay.com/browse/LPS-168262

**Comment by matuzalemsteles** [comment]
Well I'm not sure why it's not announcing correctly but the properties and focus are correct in ClayModal, the same way bootstrap is doing it we are doing it too. I'll look into what might be going on.

**Comment by marcoscv-work** [comment]
Reopening just to no forget "The announce that dialog is already opened".
Looks like a problem with an ID or something like that @matuzalemsteles already mentioned

**Comment by marcoscv-work** [comment]
Looks like fixed in clay in some way.. :-D
I will test in DXP when new clay version is merged in DXP, If needed we will open a new one
